lib/App/Pinto/Command/pull.pm

```diff
@@ -203,6 +204,51 @@ to work on those distributions, in the event that you need to patch them
 in the future.  Be aware that most distributions do not actually declare
 their development prerequisites.
 
+=item --verify-upstream=[0-5]
+
+=item -Z [0-5]
+
+=item -Z ... -ZZZZZ
+
+!! THIS OPTION IS EXPERIMENTAL !!
+
+Require upstream distribution files to be verified before operating on them.
+Repeated use of this argument (up to 5) requires the upstream verification to
+be progressively more strict.  You can also set the verification level
+explicitly, e.g.,
+
+    --verify-upstream=3
+
+At level 0, no verification is performed. This may be useful if you need to
+override the verification level set earlier, say, in a script.
+
+At level 1, we verify the distributions checksum using the upstream CHECKSUMS
+file.
+
+At level 2, we also verify the signature on the upstream CHECKSUMS file if it
+has one.  Warnings about unknown or untrusted PGP keys relating to that file
+are printed.
+
+At level 3, we also require upstream CHECKSUMS files to be signed.  Warnings
+about unknown or untrusted PGP keys relating to that file are now considered
+fatal.
+
+At level 4, we also verify the unpacked distribution using the embedded
```

Comment:
Yeah, that's not ideal. I could not figure out how to push it down that far
without breaking things. Let me know if you have any ideas :-)


Kahlil (Kal) Hodgson                       GPG: C9A02289
Head of Technology                         (m) +61 (0) 4 2573 0382
DealMax Pty Ltd

Suite 1416
401 Docklands Drive
Docklands VIC 3008 Australia

"All parts should go together without forcing.  You must remember that
the parts you are reassembling were disassembled by you.  Therefore,
if you can't get them together again, there must be a reason.  By all
means, do not use a hammer."  -- IBM maintenance manual, 1925

On 6 February 2015 at 16:23, Jeffrey Ryan Thalhammer <
notifications@github.com> wrote:

> In lib/App/Pinto/Command/pull.pm
> <https://github.com/thaljef/Pinto/pull/182#discussion_r24223287>:
>
> > +
> > +At level 0, no verification is performed. This may be useful if you need to
> > +override the verification level set earlier, say, in a script.
> > +
> > +At level 1, we verify the distributions checksum using the upstream CHECKSUMS
> > +file.
> > +
> > +At level 2, we also verify the signature on the upstream CHECKSUMS file if it
> > +has one.  Warnings about unknown or untrusted PGP keys relating to that file
> > +are printed.
> > +
> > +At level 3, we also require upstream CHECKSUMS files to be signed.  Warnings
> > +about unknown or untrusted PGP keys relating to that file are now considered
> > +fatal.
> > +
> > +At level 4, we also verify the unpacked distribution using the embedded
>
> Does that mean if we pull --verify-upstream=4 that the distro will be
> unpacked twice (once to parse for packages, and once to check the
> signatures)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/thaljef/Pinto/pull/182/files#r24223287>.
>